### ISSUE TITLE
 Perhaps support cargo-generate

### ISSUE THREAD

**Issue by janhohenheim** [TheBevyFlock/bevy-template#1]
Migrated from https://github.com/bevyengine/bevy_github_ci_template/issues/59

**Comment by Sapein** [comment]
Either way, I do think supporting cargo generate would mostly provide some nice features while not necessarily requiring or taking away too much from just using this repository as a github template if we wanted to do that. 

Also, Would this be better suited for a discussion? or should we keep this as an issue?

**Comment by janhohenheim** [comment]
I think for small repos, having only issues keeps stuff more visible :)
Agreed, cargo generate would be very nice to have.

**Comment by Sapein** [comment]
Alright, so next question is: How much do we want to integrate with cargo-generate? Just basic integration, or do we want 'deeper' integration (such as, using conditionals and the like). If we go with deeper integration, we can do a lot more with it, BUT using the repo as a 'Github Repository Template' would be more tricky/less intuitive. Where-as if we keep it rather basic, then using this as a 'Github Repository Template' would have no issues.

**Comment by benfrankel** [comment]
IMO cargo-generate support should be added after #19, since some hack may be required to make the two features compatible, and having an itch.io page ASAP will be useful for testing.

**Comment by Sapein** [comment]
After doing some research, it seems like `cargo-generate` does not choke when ran on a repository that doesn't have anything in it. I also found a few different repositories that use github actions:

- https://github.com/rust-github/template/tree/main
- https://github.com/fission-codes/rust-template
- https://github.com/macalimlim/godot-rust-template
- https://github.com/SrTemplates/Bevy

Not all of these use github actions directly (at least two support it, but don't run it on the main repo it seems), while two do. So we can probably look at these to see how they operate and how to integrate them into what we need.

**Comment by MiniaczQ** [comment]
Out of scope for now
https://discord.com/channels/691052431525675048/1258521739395203174/1260625766572359790

**Comment by Sapein** [comment]
> Out of scope for 0.14 template Will return to this topic when editor lands, see: https://discord.com/channels/691052431525675048/1258521739395203174/1260625766572359790

No it isn't. That was more-so talking about Bevy itself from what I could tell. This isn't bevy.

**Comment by MiniaczQ** [comment]
Agreed in discord to maintain a `cargo-generate` branch which has the template variables

**Comment by janhohenheim** [comment]
Done!
Run `cargo generate TheBevyFlock/bevy-template --branch cargo-generate` :)
